app-sso/how-to-guides/service-operators/verify-token-settings.hbs.md

```diff
@@ -0,0 +1,192 @@
+# Verify token settings
+
+After you set up an AppSSO Authserver, you may want to verify that the token applications receive looks as expected.
+To do this, you can create a simple application consuming your Authserver. The following `yaml` creates such an application
+that, when you visit its url, lets you log in using your Authserver and then displays the token it receives.
+
+>**Note:** that this is **not** intended to be an app to be run in production, but rather a tool to help you verify your
+> setup.
+
+Assuming you saved the below yaml in a file called `token-viewer.yaml`, you can apply this to your cluster with the following command:
+
+```shell
+ytt -f token-viewer.yaml --data-value ingress_domain=<Your ingress domain> --data-value-yaml 'authserver_selector=<Your authserver selector>' | kubectl apply -f-
+```
+
+The authserver selector field is populated by adding a label name and its value. For example: `{"name": "ci"}`
+If you need more help understanding the Authserver selector field, please read more here.
+
+```yaml
+#!
+#! Token viewer
+#!
+#! usage:
+#!
+#! ytt -f client.yml --data-value ingress_domain=example.com --data-value-yaml 'authserver_selector={"name": "ci"}'
+#!
+#! Then navigate to http://token-viewer.<INGRESS_DOMAIN>
+#!
+#@ load("@ytt:data", "data")
+#@ fqdn = "token-viewer." + data.values.ingress_domain
+#@ redirect_uri = "http://" + fqdn + "/oauth2/callback"
+---
+apiVersion: sso.apps.tanzu.vmware.com/v1alpha1
+kind: ClientRegistration
+metadata:
+   name: my-client-registration
+   namespace: default
+spec:
+   authServerSelector:
+      matchLabels: #@ data.values.authserver_selector
+   redirectURIs:
+      - #@ redirect_uri
+   requireUserConsent: false
+   clientAuthenticationMethod: client_secret_basic
+   authorizationGrantTypes:
+      - "authorization_code"
+   scopes:
+      - name: "openid"
+      - name: "email"
+      - name: "profile"
+      - name: "roles"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-application
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: test-application
+  template:
+    metadata:
+      labels:
+        name: test-application
+    spec:
+      containers:
+        - image: bitnami/oauth2-proxy:7.3.0
```

Comment:
**[thought, non-blocking]:** I wonder if we should include a note/warning that this is pulling in an unvetted, public image even though its bitnami. Also, it won't work behind an air-gap.

Reply:
Great call! Yes, we definitely should include a warning. And yes, I will also add the air gap note.